shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -158,8 +158,13 @@ - (BOOL)application:(UIApplication*)application
                        FML_LOG(ERROR)
                            << "Timeout waiting for the first frame when launching an URL.";
                      } else {
+                       NSString* pathAndQuery = url.path;
+                       if ([url.query length] != 0) {
+                         pathAndQuery =
+                             [NSString stringWithFormat:@"%@?%@", pathAndQuery, url.query];
+                       }
                        [flutterViewController.engine.navigationChannel invokeMethod:@"pushRoute"
-                                                                          arguments:url.path];
+                                                                          arguments:pathAndQuery];
```

Comment:
This could also just be `[uri absoluteString]` right? Instead of having to construct the URL string manually.